packages/vite/src/client/client.ts

```diff
@@ -546,10 +546,10 @@ export function createHotContext(ownerPath: string): ViteHotContext {
     // eslint-disable-next-line @typescript-eslint/no-empty-function
     decline() {},
 
+    // tell the server to re-perform hmr propagation from this module as root
     invalidate() {
-      // TODO should tell the server to re-perform hmr propagation
-      // from this module as root
-      location.reload()
+      notifyListeners('vite:invalidate', ownerPath)
```

Comment:
Sapphi is talking about this client-side event, I think

Reply:
I wonder if this client event is even worth adding? What's the use case?

Reply:
I assume the use case for all of these is to help debug HMR problems and also to confirm they're working as expected in tests. 

If I add the type to the HMRPayload union, then I'd need to handle it where I pointed out because of the exhaustive switch statement. I could create a separate type that's not included in that union, but it would be different from the others.  But, I guess it is different, so maybe that's fine. 

I'm also happy to remove the notifyListeners if that's preferred.